### PR TITLE
[@types/google-maps-react] Make Marker and InfoWindow components extandable.

### PR DESCRIPTION
**Pull request by MityaSavchenko**
Add missing event handlers for marker props.
Add missing methods and variables for Marker and InfoWindow classes.

### CODE REVIEW

File: types/google-maps-react/index.d.ts [hunk 1]
```diff
@@ -94,15 +103,21 @@ export class HeatMap extends React.Component<any, any> {
 }
 
 export interface InfoWindowProps extends Partial<google.maps.InfoWindowOptions> {
-  google: typeof google;
-  map: google.maps.Map;
-  marker: google.maps.Marker;
+  google?: typeof google;
```
Comment (gburgett) [inline types/google-maps-react/index.d.ts]: I believe these properties are intended to be required.  Would you please look up the documentation and determine if they are optional?

File: types/google-maps-react/index.d.ts [hunk 2]
```diff
@@ -71,14 +71,23 @@ export interface MarkerProps extends Partial<google.maps.MarkerOptions> {
...
+  onMouseup?: markerEventHandler;
+  onRecenter?: markerEventHandler;
 }
 
 export class Map extends React.Component<MapProps, any> {
 }
 
-export class Marker extends React.Component<MarkerProps, any> {
+export class Marker<P extends MarkerProps = MarkerProps, S = any> extends React.Component<P, S> {
+  marker?: google.maps.Marker;
```
Comment (gburgett) [inline types/google-maps-react/index.d.ts]: When you create a `Marker` class, is it possible for this to not exist?

**Review by gburgett — CHANGES_REQUESTED** [review]
Please double check on those props, whether they are optional or required.

File: types/google-maps-react/index.d.ts (hunk 1 above)
Comment (MityaSavchenko) [reply types/google-maps-react/index.d.ts]: And should i fix it?

Comment (MityaSavchenko) [reply types/google-maps-react/index.d.ts]: > I believe these properties are intended to be required. Would you please look up the documentation and determine if they are optional?
And should i fix it?

File: types/google-maps-react/index.d.ts (hunk 2 above)
Comment (gburgett) [reply types/google-maps-react/index.d.ts]: Ok this one's fine then.

File: types/google-maps-react/index.d.ts (hunk 1 above)
Comment (gburgett) [reply types/google-maps-react/index.d.ts]: Right, you're supposed to pass them like this:

```tsx
public render() {
  return <div>
    <InfoWindow google={google} map={map} marker={marker} />
  </div>
}
```

And Typescript ought to tell you that you're missing one of them.

Comment (MityaSavchenko) [reply types/google-maps-react/index.d.ts]: Okey. Thanks. Fixed that

**Review by gburgett — APPROVED** [review]

**Comment by gburgett** [comment]
@MityaSavchenko Thanks for addressing my concerns!

**Pull request merged**